src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,40 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    static final int FIGURES_COUNT = 5;
+    static final int DEFAULT_RADIUS = 10;
+    static final int MAX_RANDOM_VALUE = 25;
+
+    public Figure getRandomFigure() {
+        Random randomFigure = new Random();
+        switch (randomFigure.nextInt(FIGURES_COUNT)) {
+            case (0):
+                return new Circle(new ColorSupplier().getRandomColor(),
+                        new Random().nextInt(MAX_RANDOM_VALUE));
```

Comment:
let's create `new ColorSupplier()` and `new Random()` only once - use local variables